### PR TITLE
[SPARK-49993][SQL] Improve error messages for Sum and Average

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
This PR improves messages for ANSI related issues for Sum and Average.

### Why are the changes needed?
The [PR](https://github.com/apache/spark/pull/48206) for removing ANSI suggestion in ARITHMETIC_OVERFLOW was getting too big, so this PR aims to split the work into multiple tasks.

### Does this PR introduce _any_ user-facing change?
Yes, new suggestions are added to do try_sum and try_average

### How was this patch tested?
Tests added.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/results/ansi/math.sql.out [hunk 1]
```diff
@@ -761,7 +761,7 @@ org.apache.spark.SparkArithmeticException
   "errorClass" : "ARITHMETIC_OVERFLOW",
   "sqlState" : "22003",
   "messageParameters" : {
-    "alternative" : "",
+    "alternative" : " Use 'try_conv' to tolerate overflow and return NULL instead.",
```
Comment (srielau) [inline sql/core/src/test/resources/sql-tests/results/ansi/math.sql.out:764]: Fundamental: Why do we need to put a sentence into alternative? Is there a way to put the nglish portion into the message text?

**Review by srielau — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala [hunk 2]
```diff
@@ -418,6 +418,47 @@ case class Acosh(child: Expression)
...
+    numExpr: Expression,
+    fromBaseExpr: Expression,
+    toBaseExpr: Expression,
+    replacement: Expression)
+  extends RuntimeReplaceable with InheritAnalysisRules {
+
+  def this(
+            numExpr: Expression,
+            fromBaseExpr: Expression,
+            toBaseExpr: Expression) =
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala:449]: Please, fix indentations.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala [hunk 3]
```diff
@@ -418,6 +418,47 @@ case class Acosh(child: Expression)
...
+ * @param toBaseExpr to which base
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(num, from_base, to_base) - Convert `num` from `from_base` to `to_base`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('100', 2, 10);
+       4
+      > SELECT _FUNC_(-10, 16, -10);
+       -16
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala:435]: I believe we should show at least one example of the `NULL` result.

File: python/pyspark/sql/functions/builtin.py [hunk 4]
```diff
@@ -7248,6 +7248,41 @@ def log2(col: "ColumnOrName") -> Column:
...
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        logariphm of given value.
+
+    Examples
+    --------
+    >>> df = spark.createDataFrame([("010101",)], ['n'])
+    >>> df.select(try_conv(df.n, 2, 16).alias('hex')).collect()
+    [Row(hex='15')]
```
Comment (MaxGekk) [inline python/pyspark/sql/functions/builtin.py:7277]: can you show an example of NULL result?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala [hunk 5]
```diff
@@ -270,7 +272,8 @@ case class DecimalDivideWithOverflowCheck(
       if (nullOnOverflow)  {
         null
       } else {
-        throw QueryExecutionErrors.overflowInSumOfDecimalError(getContextOrNull())
+        throw QueryExecutionErrors.overflowInSumOfDecimalError(getContextOrNull(),
+          suggestedFunc = "try_average")
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala:276]: Is there this functions? I have found only `try_avg`:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L462

**Review by HyukjinKwon — COMMENTED** [review]
Let's add a unittest in, e.g., test_functions.py so the test can be shared between connect and classic

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailom-db and @srielau @HyukjinKwon for review.

**Pull request closed without merging**